pkg/cmd/server/election/lease_endpoint_reconciler.go

```diff
@@ -174,7 +174,12 @@ func (r *leaseEndpointReconciler) ReconcileEndpoints(serviceName string, ip net.
 	}
 
 	glog.Warningf("Resetting endpoints for master service %q to %v", serviceName, masterIPs)
-	return r.endpointRegistry.UpdateEndpoints(ctx, e)
+	return r.endpointRegistry.UpdateEndpoints(ctx, e, nil, nil)
+}
+
+// TODO collapse onto upstream
+func (r *leaseEndpointReconciler) StopReconciling(serviceName string, ip net.IP, endpointPorts []api.EndpointPort) error {
```

Comment:
implications of ignoring this?

Reply:
> implications of ignoring this?

It hangs open on shutdown instead of cleanly closing.  This is the target of a post-rebase task already to collapse onto upstream.